src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkTask.java

```diff
@@ -142,8 +142,21 @@ private void ensureIndexOrDataStreamExists(final String index) {
         if (!indexCache.contains(index)) {
             if (!client.indexOrDataStreamExists(index)) {
                 if (config.dataStreamEnabled()) {
-                    LOGGER.info("Create data stream {}", index);
-                    client.createIndexTemplateAndDataStream(index, config.dataStreamTimestampField());
+                    // Create if data stream template if does not exist
+                    if (config.dataStreamExistingIndexTemplateName() != null
```

Comment:
The checks will be much more readable after https://github.com/Aiven-Open/opensearch-connector-for-apache-kafka/pull/293/files#r1705468208